### DISCUSSION TITLE
Various fixes and tweaks (Godot 4)

### DISCUSSION THREAD

**Discussion by BlueStaggo** [Techokami/SonicWorldsNext#36] [Show and tell]
Sonic Worlds Next is a revelation to me because I've always wanted an easy to use classic Sonic game framework on a FOSS game engine, but it is quite buggy at the moment. That is why I decided to implement some fixes myself! These changes include, but are not limited to:
- Fixed incorrectly looping animations (Tails' tails move correctly and Knuckles climbs properly)
- Readded the ring animation on the title screen and the missing parts of the title card
- Fixed a few of Sonic's animations being off-centered, especially his walking animation
- The player correctly hangs on bars
- Fixed a ton of warnings
- The collision of Base Zone Act 2 has been fixed 
- Water correctly renders in different resolutions

https://user-images.githubusercontent.com/71090710/231770891-129ba2c2-f3ce-42e7-a71c-9d668583e216.mp4

I would like to make a pull request with these changes. I hope to contribute to Sonic Worlds Next further!

**Comment by Renhoex** [comment]
I'm guessing that this is based on my godot 4 branch?

Just as a heads up, I'm not too familiar with how github works but this branch wasn't meant to be used publicly yet as I'm using it as a dev branch and I'm still making a lot of fixes.

**Reply by BlueStaggo** [reply]
It's still ok if I make a pull request for it, right?

**Reply by Renhoex** [reply]
> It's still ok if I make a pull request for it, right?

I'd need to review the changes to make sure there wouldn't be any conflicts, I'm already guessing there'd be some conflicting changes since I actually went through and removed a lot of the debug errors in the update I pushed today.

**Reply by BlueStaggo** [reply]
Well, I'll get to work merging your changes into my fork and make a pull request once it's done.

**Reply by Renhoex** [reply]
Sure, honestly I appreciate any help in getting the godot 4 port ready and out, I'm just very particular about changes in the code.
